Views raise exception when using SEARCH method

## Checklist

- [x] I have included information about relevant versions
- [x] I have verified that the issue persists when using the `master` branch of Faust.

## Steps to reproduce

I wrote this small Faust app:
```python
import faust

app = faust.App(
    id='api-issue',
)

@app.page('/test')
async def test(web, request):
    return web.text('Working!')
```

App was started successfully:
```bash
➜ pipenv run faust -A app worker -l info
┌ƒaµS† v1.8.1─┬────────────────────────────────────────────────────────────────────────┐
│ id          │ api-issue                                                              │
│ transport   │ [URL('kafka://localhost:9092')]                                        │
│ store       │ memory:                                                                │
│ web         │ http://localhost:6066/                                                 │
│ log         │ -stderr- (info)                                                        │
│ pid         │ 95220                                                                  │
│ hostname    │ ip-192-168-1-179.eu-west-1.compute.internal                            │
│ platform    │ CPython 3.7.4 (Darwin x86_64)                                          │
│        +    │ Cython (Clang 10.0.1 (clang-1001.0.46.4))                              │
│ drivers     │                                                                        │
│   transport │ aiokafka=1.0.6                                                         │
│   web       │ aiohttp=3.6.2                                                          │
│ datadir     │ /Users/ignacio/dev/playground/faust-api-search-issue/api-issue-data    │
│ appdir      │ /Users/ignacio/dev/playground/faust-api-search-issue/api-issue-data/v1 │
└─────────────┴────────────────────────────────────────────────────────────────────────┘
[2019-10-29 14:36:19,862] [95220] [INFO] [^Worker]: Starting...
[2019-10-29 14:36:19,865] [95220] [INFO] [^-App]: Starting...
[2019-10-29 14:36:19,866] [95220] [INFO] [^--Monitor]: Starting...
[2019-10-29 14:36:19,866] [95220] [INFO] [^--Producer]: Starting...
[2019-10-29 14:36:19,866] [95220] [INFO] [^---ProducerBuffer]: Starting...
[2019-10-29 14:36:19,886] [95220] [INFO] [^--CacheBackend]: Starting...
[2019-10-29 14:36:19,886] [95220] [INFO] [^--Web]: Starting...
[2019-10-29 14:36:19,886] [95220] [INFO] [^---Server]: Starting...
[2019-10-29 14:36:19,887] [95220] [INFO] [^--Consumer]: Starting...
[2019-10-29 14:36:19,888] [95220] [INFO] [^---AIOKafkaConsumerThread]: Starting...
[2019-10-29 14:36:20,869] [95220] [INFO] [^--LeaderAssignor]: Starting...
[2019-10-29 14:36:20,870] [95220] [INFO] [^--Producer]: Creating topic 'api-issue-__assignor-__leader'
[2019-10-29 14:36:20,918] [95220] [INFO] [^--Producer]: Topic 'api-issue-__assignor-__leader' created.
[2019-10-29 14:36:20,919] [95220] [INFO] [^--ReplyConsumer]: Starting...
[2019-10-29 14:36:20,919] [95220] [INFO] [^--AgentManager]: Starting...
[2019-10-29 14:36:20,919] [95220] [INFO] [^--Conductor]: Starting...
[2019-10-29 14:36:20,919] [95220] [INFO] [^--TableManager]: Starting...
[2019-10-29 14:36:21,922] [95220] [INFO] [^---Recovery]: Starting...
[2019-10-29 14:36:22,925] [95220] [INFO] [^--Producer]: Creating topic 'api-issue-__assignor-__leader'
[2019-10-29 14:36:23,912] [95220] [INFO] Updating subscribed topics to: frozenset({'api-issue-__assignor-__leader'})
[2019-10-29 14:36:23,913] [95220] [INFO] Subscribed to topic(s): {'api-issue-__assignor-__leader'}
[2019-10-29 14:36:23,926] [95220] [INFO] Discovered coordinator 1001 for group api-issue
[2019-10-29 14:36:23,927] [95220] [INFO] Revoking previously assigned partitions set() for group api-issue
[2019-10-29 14:36:24,919] [95220] [INFO] (Re-)joining group api-issue
[2019-10-29 14:36:24,927] [95220] [INFO] Joined group 'api-issue' (generation 1) with member_id faust-1.8.1-d689b013-fba6-4051-ac15-5eabc1886e46
[2019-10-29 14:36:24,927] [95220] [INFO] Elected group leader -- performing partition assignments using faust
[2019-10-29 14:36:24,938] [95220] [INFO] Successfully synced group api-issue with generation 1
[2019-10-29 14:36:24,939] [95220] [INFO] Setting newly assigned partitions {TopicPartition(topic='api-issue-__assignor-__leader', partition=0)} for group api-issue
[2019-10-29 14:36:25,493] [95220] [INFO] [^---Recovery]: Resuming flow...
[2019-10-29 14:36:25,494] [95220] [INFO] [^---Recovery]: Seek stream partitions to committed offsets.
[2019-10-29 14:36:25,881] [95220] [INFO] [^--Fetcher]: Starting...
[2019-10-29 14:36:25,881] [95220] [INFO] [^---Recovery]: Worker ready
[2019-10-29 14:36:25,882] [95220] [INFO] [^Worker]: Ready
```

Then call the view using `SEARCH` method:
```bash
➜ curl -XSEARCH localhost:6066
```

## Expected behavior

I expected to behave as other methods, example below when using `PUT`:
```
➜ curl -XPUT localhost:6066
{"error": "Method PUT not allowed."}
```

## Actual behavior

Faust raised and exception:
```
[2019-10-29 14:36:47,703] [95220] [ERROR] Error handling request
Traceback (most recent call last):
  File "/Users/ignacio/.virtualenvs/faust-api-search-issue-6iz8n7Lc/lib/python3.7/site-packages/aiohttp/web_protocol.py", line 418, in start
    resp = await task
  File "/Users/ignacio/.virtualenvs/faust-api-search-issue-6iz8n7Lc/lib/python3.7/site-packages/aiohttp/web_app.py", line 458, in _handle
    resp = await handler(request)
  File "/Users/ignacio/.virtualenvs/faust-api-search-issue-6iz8n7Lc/lib/python3.7/site-packages/faust/web/drivers/aiohttp.py", line 241, in _dispatch
    return await handler(request)
  File "/Users/ignacio/.virtualenvs/faust-api-search-issue-6iz8n7Lc/lib/python3.7/site-packages/faust/web/views.py", line 76, in __call__
    return await self.dispatch(request)
  File "/Users/ignacio/.virtualenvs/faust-api-search-issue-6iz8n7Lc/lib/python3.7/site-packages/faust/web/views.py", line 87, in dispatch
    method = cast(Callable[..., Awaitable[Response]], self.methods[method])
KeyError: 'search'
```

When I call the view using `SEARCH` method, I got the following:
```bash
➜ curl -XSEARCH localhost:6066
500 Internal Server Error

Server got itself in trouble%
```

## Full traceback

```pytb
[2019-10-29 14:36:47,703] [95220] [ERROR] Error handling request
Traceback (most recent call last):
  File "/Users/ignacio/.virtualenvs/faust-api-search-issue-6iz8n7Lc/lib/python3.7/site-packages/aiohttp/web_protocol.py", line 418, in start
    resp = await task
  File "/Users/ignacio/.virtualenvs/faust-api-search-issue-6iz8n7Lc/lib/python3.7/site-packages/aiohttp/web_app.py", line 458, in _handle
    resp = await handler(request)
  File "/Users/ignacio/.virtualenvs/faust-api-search-issue-6iz8n7Lc/lib/python3.7/site-packages/faust/web/drivers/aiohttp.py", line 241, in _dispatch
    return await handler(request)
  File "/Users/ignacio/.virtualenvs/faust-api-search-issue-6iz8n7Lc/lib/python3.7/site-packages/faust/web/views.py", line 76, in __call__
    return await self.dispatch(request)
  File "/Users/ignacio/.virtualenvs/faust-api-search-issue-6iz8n7Lc/lib/python3.7/site-packages/faust/web/views.py", line 87, in dispatch
    method = cast(Callable[..., Awaitable[Response]], self.methods[method])
KeyError: 'search'
```

# Versions

* Python version: 3.7
* Faust version: 1.8.1
* Operating system: macOS Mojave 10.14.6

--- a/faust/web/views.py
+++ b/faust/web/views.py
@@ -64,6 +64,7 @@ class View:
             'delete': self.delete,
             'put': self.put,
             'options': self.options,
+            'search': self.search,
         }
         self.__post_init__()
 
@@ -161,6 +162,11 @@ class View:
     async def options(self, request: Request, **kwargs: Any) -> Any:
         """Override ``options`` to define the HTTP OPTIONS handler."""
         raise exceptions.MethodNotAllowed('Method OPTIONS not allowed.')
+
+    @no_type_check  # subclasses change signature based on route match_info
+    async def search(self, request: Request, **kwargs: Any) -> Any:
+        """Override ``search`` to define the HTTP SEARCH handler."""
+        raise exceptions.MethodNotAllowed('Method SEARCH not allowed.')
 
     def text(self, value: str, *,
              content_type: str = None,

--- a/t/unit/web/test_views.py
+++ b/t/unit/web/test_views.py
@@ -37,11 +37,12 @@ class test_View:
             'delete': view.delete,
             'put': view.put,
             'options': view.options,
+            'search': view.search,
         }
 
     @pytest.mark.parametrize('method', [
-        'GET', 'POST', 'PATCH', 'DELETE', 'PUT', 'OPTIONS',
-        'get', 'post', 'patch', 'delete', 'put', 'options',
+        'GET', 'POST', 'PATCH', 'DELETE', 'PUT', 'OPTIONS', 'SEARCH',
+        'get', 'post', 'patch', 'delete', 'put', 'options', 'search',
     ])
     @pytest.mark.asyncio
     async def test_dispatch(self, method, *, view):
@@ -102,6 +103,11 @@ class test_View:
     async def test_interface_options(self, *, view):
         with pytest.raises(MethodNotAllowed):
             await view.options(Mock(name='request', autospec=Request))
+
+    @pytest.mark.asyncio
+    async def test_interface_search(self, *, view):
+        with pytest.raises(MethodNotAllowed):
+            await view.search(Mock(name='request', autospec=Request))
 
     def test_text(self, *, view, web):
         response = view.text(

